lib/posix/options/pthread.c

```diff
@@ -384,6 +384,40 @@ int pthread_attr_setstack(pthread_attr_t *_attr, void *stackaddr, size_t stacksi
 	return 0;
 }
 
+/**
+ * @brief Get scope attributes in thread attributes object.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_attr_getscope(const pthread_attr_t *_attr, int *contentionscope)
+{
+	struct posix_thread_attr *attr = (struct posix_thread_attr *)_attr;
+
+	if (!__attr_is_initialized(attr) || contentionscope == NULL) {
+		return EINVAL;
+	}
+	*contentionscope = attr->contentionscope;
+	return 0;
+}
+
+/**
+ * @brief Set scope attributes in thread attributes object.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_attr_setscope(pthread_attr_t *_attr, int contentionscope)
+{
+	struct posix_thread_attr *attr = (struct posix_thread_attr *)_attr;
+
+	if (!__attr_is_initialized(attr) ||
+	    (contentionscope != PTHREAD_SCOPE_SYSTEM && contentionscope != PTHREAD_SCOPE_PROCESS)) {
+		LOG_ERR("Invalid pthread_attr_t or contentionscope");
+		return EINVAL;
+	}
+	attr->contentionscope = contentionscope;
```

Comment:
Just reading up on this a bit more
 https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_attr_getscope.html

I think `pthread_attr_init()` should be modified  such that `contentionscope = PTHREAD_SCOPE_SYSTEM`.

This function should likely return `ENOTSUP` if `*contentionscope == PTHREAD_SCOPE_PROCESS` because Zephyr does not support per-process thread scheduling.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_09_04